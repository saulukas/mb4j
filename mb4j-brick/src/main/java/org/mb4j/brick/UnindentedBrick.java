package org.mb4j.brick;

import java.io.Writer;
import org.mb4j.brick.jmustache.Template.Fragment;
import org.mb4j.brick.renderer.RenderingScope;

public class UnindentedBrick extends MustacheBrick {

    @Override
    public void execute(Fragment frag, Writer out) {
        ((RenderingScope) out).renderUnindented(this, frag);
    }
}
