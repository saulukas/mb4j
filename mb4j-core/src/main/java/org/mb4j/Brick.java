package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Mustache.Lambda;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class Brick implements Lambda {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderIndented(this, frag);
  }
}
