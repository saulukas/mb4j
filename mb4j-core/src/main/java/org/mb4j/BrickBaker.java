package org.mb4j;

public interface BrickBaker<P> {
  Brick bakeBrick(P params);
}
