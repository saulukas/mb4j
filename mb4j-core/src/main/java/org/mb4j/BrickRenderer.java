package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import org.mb4j.template.TemplateProvider;
import java.io.Writer;

public class BrickRenderer {
  private final TemplateProvider templateProvider;

  public BrickRenderer(TemplateProvider templateProvider) {
    this.templateProvider = templateProvider;
  }

  public void render(Brick brick, Writer out) {
    RenderingScope scope = new RenderingScope(out, templateProvider);
    scope.render(brick);
  }
}
