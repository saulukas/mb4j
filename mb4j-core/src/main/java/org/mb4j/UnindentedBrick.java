package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class UnindentedBrick extends Brick {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderUnindented(this, frag);
  }
}
