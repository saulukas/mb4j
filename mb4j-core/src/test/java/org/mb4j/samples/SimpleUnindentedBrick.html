
<pre>
  
    Some simple ASCII form
    ----------------------

    Name    : {{name}}
    Address : {{address}}

</pre>