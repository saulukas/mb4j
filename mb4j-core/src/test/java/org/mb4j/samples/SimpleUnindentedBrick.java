package org.mb4j.samples;

import org.mb4j.UnindentedBrick;

public class SimpleUnindentedBrick extends UnindentedBrick {
  String name = "JokÅ«bas";
  String address = "Vilnius, Lithuania";
}
