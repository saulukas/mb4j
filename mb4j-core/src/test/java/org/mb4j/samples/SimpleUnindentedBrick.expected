
<pre>
  
    Some simple ASCII form
    ----------------------

    Name    : JokÅ«bas
    Address : Vilnius, Lithuania

</pre>